src/Types/TxConstraints.purs

```diff
@@ -186,6 +196,56 @@ mustBeSignedBy = singleton <<< MustBeSignedBy
 mustIncludeDatum :: forall (i :: Type) (o :: Type). Datum -> TxConstraints i o
 mustIncludeDatum = singleton <<< MustIncludeDatum
 
+-- | Require the transaction to include a reference input.
+mustIncludeReferenceInput
```

Comment:
I thought we were going to wait to add the new constraints until after the v2.0.0 release, but I'm reconsidering that now
- you've already implemented these
- the changes are largely backwards-compatible and unobtrusive
- we have a growing number of projects who want to use the new features (at least three for sure)

So I think we should keep the new constraints. We just need to deal with `Datum` arguments to other ones (not related to this PR)